internal/statemachine/snap.go

```diff
@@ -56,5 +61,50 @@ func (snapStateMachine *SnapStateMachine) Setup() error {
 		return err
 	}
 
+	return snapStateMachine.SetSeries()
+}
+
+func (snapStateMachine *SnapStateMachine) SetSeries() error {
+	model, err := snapStateMachine.decodeModelAssertion()
+	if err != nil {
+		return err
+	}
+
+	// we extracted a "core" series, containing only the major version, not minor
+	// In this case we will always end up using the LTS, so .04 minor version.
+	snapStateMachine.series = fmt.Sprintf("%s.04", model.Series())
+
 	return nil
 }
+
+// Copied and adapted from snapd/image/image_linux.go
+
+// these are postponed, not implemented or abandoned, not finalized,
+// don't let them sneak in into a used model assertion
+var reserved = []string{"core", "os", "class", "allowed-modes"}
+
+func (snapStateMachine *SnapStateMachine) decodeModelAssertion() (*asserts.Model, error) {
+	fn := snapStateMachine.Args.ModelAssertion
+
+	rawAssert, err := os.ReadFile(fn)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read model assertion: %s", err)
+	}
+
+	ass, err := asserts.Decode(rawAssert)
```

Comment:
That's a bit of an unfavorable name I must say..! I'd prefer if we change it to `assertion` or something ;)

Reply:
Yes! I did not even notice (see previous comment).